cmd/platform/runner/resync.go

```diff
@@ -0,0 +1,62 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cmdrunner
+
+import (
+	"fmt"
+
+	"github.com/elastic/cloud-sdk-go/pkg/util/cmdutil"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/ecctl/pkg/ecctl"
+	"github.com/elastic/ecctl/pkg/platform/runner"
+)
+
+var resyncRunnerCmd = &cobra.Command{
+	Use:     "resync {<runner id> | --all}",
+	Short:   "Resynchronizes the search index and cache for the selected runner or all",
+	PreRunE: cmdutil.CheckInputHas1ArgsOr0ArgAndAll,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		all, _ := cmd.Flags().GetBool("all")
+
+		if all {
+			fmt.Println("Resynchronizing all runners")
+			res, err := runner.ResyncAll(runner.Params{
+				API: ecctl.Get().API,
+			})
+			if err != nil {
+				return err
+			}
+
+			return ecctl.Get().Formatter.Format("", res)
+		}
+
+		fmt.Printf("Resynchronizing runner: %s\n", args[0])
+		return runner.Resync(runner.ResyncParams{
```

Comment:
I wonder if it's intentional to only return an error here and a payload and an error in the call for `runner.ResyncAll`

Reply:
Agree I don't think this is what we are doing in other commands, do we?

Reply:
Heya! yeah, it's intentional. When you resync one runner the output is
```
{
}
```
 But when you resync --all the output is a list of the runners which where resynchronised. I thought it was kind of ugly/confusing to have just a `{}` as the output. Happy to change that if everyone disagrees with me though 😄